### PR TITLE
PYIC-2960: Review consistency of log messages

### PR DESCRIPTION
**Pull request by TalatAydinGDS**
### What changed

-  All lambdas were checked and it was verified they all use PowerTools logging and PowerToolsTracing for AOP.Also, Lib services were checked and it was verified they all use PowerTools logging.

- All Lambdas have been adjusted to the same field. 

- All log levels in the entire project have been reviewed. Necessary changes have been made where needed.

-  The exceptions handled by Lambdas are now written in a similar format with log display. 

-  The new log fields have been added, and they are now being called from all relevant places in a similar format.

-  All log config files within all lambdas were reviewed and compared.

-  The CriId locations where the log is written were checked within the project. 

### Why did it change

  Review consistency of log messages

### Issue tracking

- [PYIC-2960](https://govukverify.atlassian.net/browse/PYIC-2960)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-2960]: https://govukverify.atlassian.net/browse/PYIC-2960?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lambdas/build-client-oauth-response/src/main/java/uk/gov/di/ipv/core/buildclientoauthresponse/BuildClientOauthResponseHandler.java [hunk 1]
```diff
@@ -188,35 +188,32 @@ public Map<String, Object> handleRequest(JourneyRequest input, Context context)
 
             return clientResponse.toObjectMap();
         } catch (ParseException e) {
-            LOGGER.error("Authentication request could not be parsed", e);
-            return new JourneyErrorResponse(
-                            JOURNEY_ERROR_PATH,
-                            HttpStatus.SC_BAD_REQUEST,
-                            ErrorResponse.FAILED_TO_PARSE_OAUTH_QUERY_STRING_PARAMETERS)
-                    .toObjectMap();
+            LogHelper.logExceptionDetails("Authentication request could not be parsed", e);
```
Comment (Wynndow) [inline lambdas/build-client-oauth-response/src/main/java/uk/gov/di/ipv/core/buildclientoauthresponse/BuildClientOauthResponseHandler.java]: This new helper seems like it's doing a similar thing to the `LogHelper.logErrorMessage` method used in the next catch block. Is it worth standardising on one?

File: lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java [hunk 2]
```diff
@@ -271,38 +272,24 @@ public Map<String, Object> handleRequest(JourneyRequest event, Context context)
...
-                                        LOG_MESSAGE_DESCRIPTION.getFieldName(),
-                                        "F2F cri pending verification."));
+                LogHelper.logMessage(Level.INFO, "F2F cri pending verification.");
                 return JOURNEY_PENDING;
             }
             case CriResponseService.STATUS_ERROR -> {
-                LOGGER.info(
-                        new StringMapMessage()
-                                .with(LOG_MESSAGE_DESCRIPTION.getFieldName(), "F2F cri error"));
+                LogHelper.logMessage(Level.INFO, "F2F cri error");
```
Comment (Wynndow) [inline lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java]: Should this be a WARN?

File: lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java [hunk 3]
```diff
@@ -331,18 +313,13 @@ private Map<String, Object> buildNotCorrelatedResponse(AuditEventUser auditEvent
...
 
     private Map<String, Object> buildF2FNoMatchResponse(AuditEventUser auditEventUser)
             throws SqsException {
-        var message =
-                new StringMapMessage()
-                        .with(
-                                LOG_MESSAGE_DESCRIPTION.getFieldName(),
-                                "F2F return - failed to match a profile.");
-        LOGGER.info(message);
+        LogHelper.logMessage(Level.WARN, "F2F return - failed to match a profile.");
```
Comment (Wynndow) [inline lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java]: Should this be info?

File: lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java [hunk 4]
```diff
@@ -356,12 +333,7 @@ private Map<String, Object> buildF2FNoMatchResponse(AuditEventUser auditEventUse
     private Map<String, Object> buildNoMatchResponse(
             List<SignedJWT> credentials, AuditEventUser auditEventUser) throws SqsException {
         if (!credentials.isEmpty()) {
-            var message =
-                    new StringMapMessage()
-                            .with(
-                                    LOG_MESSAGE_DESCRIPTION.getFieldName(),
-                                    "Failed to match profile so resetting identity.");
-            LOGGER.info(message);
+            LogHelper.logMessage(Level.WARN, "Failed to match profile so resetting identity.");
```
Comment (Wynndow) [inline lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java]: Maybe info?

File: lambdas/evaluate-gpg45-scores/src/main/java/uk/gov/di/ipv/core/evaluategpg45scores/EvaluateGpg45ScoresHandler.java [hunk 5]
```diff
@@ -143,41 +145,33 @@ public Map<String, Object> handleRequest(JourneyRequest event, Context context)
...
-            LOGGER.error("Unable to determine type of credential", e);
-            return new JourneyErrorResponse(
-                            JOURNEY_ERROR_PATH,
-                            HttpStatus.SC_INTERNAL_SERVER_ERROR,
-                            ErrorResponse.FAILED_TO_DETERMINE_CREDENTIAL_TYPE)
-                    .toObjectMap();
+            LogHelper.logExceptionDetails("Unable to determine type of credential", e);
+            return buildJourneyErrorResponse(ErrorResponse.FAILED_TO_DETERMINE_CREDENTIAL_TYPE);
         } catch (SqsException e) {
             LogHelper.logErrorMessage("Failed to send audit event to SQS queue.", e.getMessage());
```
Comment (Wynndow) [inline lambdas/evaluate-gpg45-scores/src/main/java/uk/gov/di/ipv/core/evaluategpg45scores/EvaluateGpg45ScoresHandler.java:159]: Using logErrorMessage here and not logExceptionDetails as in the previous catch blocks

File: libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java [hunk 6]
```diff
@@ -174,7 +176,7 @@ public Optional<IdentityClaim> findIdentityClaim(List<VcStoreItem> vcStoreItems)
         }
 
         if (identityClaims.isEmpty()) {
-            LOGGER.warn("Failed to generate identity claim");
+            LogHelper.logMessage(Level.INFO, "Failed to empty identity claim");
```
Comment (Wynndow) [inline libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java]:
```suggestion
            LogHelper.logMessage(Level.WARN, "Failed to find any identity claims in VCs");
```

**Review by Wynndow — COMMENTED** [review]
This all looks pretty sensible. Just a few comments/questions.

File: lambdas/build-client-oauth-response/src/main/java/uk/gov/di/ipv/core/buildclientoauthresponse/BuildClientOauthResponseHandler.java (hunk 1 above)
Comment (TalatAydinGDS) [reply lambdas/build-client-oauth-response/src/main/java/uk/gov/di/ipv/core/buildclientoauthresponse/BuildClientOauthResponseHandler.java]: I added a comment in the task about that what "Some places here have the exception written in full write to the log. No changes have ben made to them in order not affect any place. In my opinion, However, can be discussed for modification in the future." But, you are right and changed method name with logErrorMessage for that.

**Review by Wynndow — APPROVED** [review]

**Pull request merged**